specification/batch/Azure.Batch/models.tsp

```diff
@@ -0,0 +1,4083 @@
+}
+
+@doc("The number of Compute Nodes in each state for a Pool.")
+@access(Access.public, "java")
+model BatchPoolNodeCounts {
+  @doc("The ID of the Pool.")
+  poolId: string;
+
+  @doc("The number of dedicated Compute Nodes in each state.")
+  dedicated?: BatchNodeCounts;
+
+  @doc("The number of Spot/Low-priority Compute Nodes in each state.")
+  lowPriority?: BatchNodeCounts;
+}
+
+@doc("The number of Compute Nodes in each Compute Node state.")
+@access(Access.public, "java")
+model BatchNodeCounts {
+  @doc("The number of Compute Nodes in the creating state.")
+  creating: int32;
+
+  @doc("The number of Compute Nodes in the idle state.")
+  idle: int32;
+
+  @doc("The number of Compute Nodes in the offline state.")
+  offline: int32;
+
+  @doc("The number of Compute Nodes in the preempted state.")
+  preempted: int32;
+
+  @doc("The count of Compute Nodes in the rebooting state.")
+  rebooting: int32;
+
+  @doc("The number of Compute Nodes in the reimaging state.")
+  reimaging: int32;
+
+  @doc("The number of Compute Nodes in the running state.")
+  running: int32;
+
+  @doc("The number of Compute Nodes in the starting state.")
+  starting: int32;
+
+  @doc("The number of Compute Nodes in the startTaskFailed state.")
+  startTaskFailed: int32;
+
+  @doc("The number of Compute Nodes in the leavingPool state.")
+  leavingPool: int32;
+
+  @doc("The number of Compute Nodes in the unknown state.")
+  `unknown`: int32;
+
+  @doc("The number of Compute Nodes in the unusable state.")
+  unusable: int32;
+
+  @doc("The number of Compute Nodes in the waitingForStartTask state.")
+  waitingForStartTask: int32;
+
+  @doc("The total number of Compute Nodes.")
+  total: int32;
+}
+
+@doc("Contains utilization and resource usage statistics for the lifetime of a Pool.")
+@access(Access.public, "java")
+model BatchPoolStatistics {
+  @doc("The URL for the statistics.")
+  url: string;
+
+  @doc("The start time of the time range covered by the statistics.")
+  startTime: utcDateTime;
+
+  @doc("The time at which the statistics were last updated. All statistics are limited to the range between startTime and lastUpdateTime.")
+  lastUpdateTime: utcDateTime;
+
+  @doc("Statistics related to Pool usage, such as the amount of core-time used.")
+  usageStats?: BatchPoolUsageStatistics;
+
+  @doc("Statistics related to resource consumption by Compute Nodes in the Pool.")
+  resourceStats?: BatchPoolResourceStatistics;
+}
+
+@doc("Statistics related to Pool usage information.")
+@access(Access.public, "java")
+model BatchPoolUsageStatistics {
+  @doc("The start time of the time range covered by the statistics.")
+  startTime: utcDateTime;
+
+  @doc("The time at which the statistics were last updated. All statistics are limited to the range between startTime and lastUpdateTime.")
+  lastUpdateTime: utcDateTime;
+
+  @doc("The aggregated wall-clock time of the dedicated Compute Node cores being part of the Pool.")
+  dedicatedCoreTime: duration;
```

Comment:
Just a confirmation: is service using e.g. `P3M` here?
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#json-durations-use-fixed-time-intervals

Reply:
not 100% sure but I believe so, this is an old definition and has been declared as a duration in our swagger for years.